code/modules/surgery/bodyparts/dismemberment.dm

```diff
@@ -417,3 +417,9 @@
 
 	update_body_parts()
 	return TRUE
+
+/// Returns a species typepath by limb_id if one matches
+/proc/species_by_id(limb_id)
+	for (var/datum/species/species_type as anything in subtypesof(/datum/species))
+		if (species_type::id == limb_id)
+			return species_type
```

Comment:
`GLOB.species_list` has species id : species typepath.
You can use that instead of adding this function.